libs/features/mas/commerce/src/wcs.js

```diff
@@ -99,25 +85,29 @@ export function Wcs({ settings }) {
                     promises.delete(offerSelectorId);
                     resolve(resolved);
                 }
-            });
-        } catch (error) {
-            // in case of 404 WCS error caused by a request with multiple osis,
-            // fallback to `fetch-by-one` strategy
-            if (error.status === 404 && options.offerSelectorIds.length > 1) {
-                log.debug('Multi-osi 404, fallback to fetch-by-one strategy');
-                await Promise.allSettled(
-                    options.offerSelectorIds.map((offerSelectorId) =>
-                        resolveWcsOffers(
-                            { ...options, offerSelectorIds: [offerSelectorId] },
-                            promises,
-                            false, // do not reject promises for missing offers, this will be done below
-                        ),
+            });  
+          }
+          // in case of 404 WCS error caused by a request with multiple osis,
+          // fallback to `fetch-by-one` strategy
+          else if (response.status === 404 && options.offerSelectorIds.length > 1) {
+            log.debug('Multi-osi 404, fallback to fetch-by-one strategy');
+            await Promise.allSettled(
+                options.offerSelectorIds.map((offerSelectorId) =>
+                    resolveWcsOffers(
+                        { ...options, offerSelectorIds: [offerSelectorId] },
+                        promises,
+                        false, // do not reject promises for missing offers, this will be done below
                     ),
-                );
-            } else {
-                log.error('Failed:', options, error);
-                message = ERROR_MESSAGE_BAD_REQUEST;
-            }
+                ),
+            );
+          } else {
+            const error = await response.text();
+            log.error(`WCS request failed. Status: ${response.status}, message: ${error}`, options);
+            message = ERROR_MESSAGE_BAD_REQUEST;
+          }
+        } catch (e) {
+          message = 'Failed to fetch WCS offer.';
```

Comment:
maybe use a constant as the line 106 above ?

Reply:
addressed